src/components/lib/posix/posix.c

```diff
@@ -134,11 +125,12 @@ cos_mmap(void *addr, size_t length, int prot, int flags, int fd, off_t offset)
 		assert(0);
 	}
 
-	void *ret = do_mmap(length);
-
-	if (ret == (void *)-1) { /* return value comes from man page */
-		printc("mmap() failed!\n");
-		assert(0);
+	addr = (void *)cos_page_bump_allocn(&parent_cinfo, length);
```

Comment:
Have you considered using `addr = (void *)cos_page_bump_allocn(&cos_defcompinfo_curr_get()->ci, length)` here instead? By using the defkernel_api, you can avoid having your own global compinfo, which makes this integrate better into the system.